### PR TITLE
select2 based species filter

### PR DESCRIPTION
**Pull request by allentran**
I didn't add the counts to the text in the search bar, the data is there/easy to do - just wanted to see what you'd think.

note from Emily - PR does now include species counts

### CODE REVIEW

File: public/js/SpeciesSearch.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        else if (treeA.text < treeB.text){
+            return -1
+        }
+        return -1
+    }
+
+    SpeciesSearch.prototype.setSpecies = function(species) {
+        this.species = Array.from(species).filter(s => s !== 'Vacant Site').sort(treeCompare);
+        selectFilter.select2({
+            placeholder: 'Filter species',
```
Comment (emillipede) [inline public/js/SpeciesSearch.js]: "Filter by species"?

Comment (allentran) [reply public/js/SpeciesSearch.js]: Sure

**Comment by emillipede** [comment]
thank you @allentran !

**Comment by emillipede** [comment]
note - still needs to be styled

**Comment by emillipede** [comment]
fixed #81

**Pull request merged**